loader/src/main/kotlin/net/weavemc/loader/util/FileManager.kt

```diff
@@ -31,17 +34,20 @@ internal object FileManager {
         val gameVersion = GameInfo.version.versionName
         val classpath = System.getProperty("java.class.path")
         if (classpath != null) {
-            operator fun String.div(other: String) =
-                this + File.separator + other
-
             val paths = classpath.split(File.pathSeparator)
             for (path in paths) {
                 // .minecraft/versions/<ver>/<ver>.jar
-                if (path.endsWith(File.separator + gameVersion / "$gameVersion.jar")) {
+                val mcLauncherPath = File.separator + buildPath(gameVersion, "$gameVersion.jar")
+                if (path.endsWith(mcLauncherPath)) {
                     return File(path)
                 }
                 // MultiMC-like maven structure
-                if (path.endsWith("com" / "mojang" / "minecraft" / gameVersion / "minecraft-$gameVersion-client.jar")) {
+                val multiMcPath = buildPath(
+                    "com", "mojang", "minecraft",
+                    gameVersion,
+                    "minecraft-$gameVersion-client.jar"
+                )
+                if (path.endsWith(multiMcPath)) {
                     return File(path)
                 }
             }
```

Comment:
Should probably extract the constants out of the for loop that do not depend on `path`, then write:
```kt
return paths.find { it.endsWith(mcLauncherPath) || it.endsWith(multiMcPath) }
    ?: fatalError("Could not find vanilla jar for version $gameVersion")
```